File: WebPage/Model1.vb
Imports System
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity
Imports System.Linq

Partial Public Class Model1
    Inherits DbContext

    Public Sub New()
        MyBase.New("name=chess")
    End Sub

    Public Overridable Property game As DbSet(Of game)
    Public Overridable Property game_moves As DbSet(Of game_moves)
    Public Overridable Property users As DbSet(Of users)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Entity(Of game)() _
            .HasMany(Function(e) e.game_moves) _
            .WithRequired(Function(e) e.game) _
            .WillCascadeOnDelete(False)

        modelBuilder.Entity(Of users)() _
            .HasMany(Function(e) e.game) _
            .WithRequired(Function(e) e.users) _
            .HasForeignKey(Function(e) e.whiteUserId) _
            .WillCascadeOnDelete(False)

        modelBuilder.Entity(Of users)() _
            .HasMany(Function(e) e.game1) _
            .WithRequired(Function(e) e.users1) _
            .HasForeignKey(Function(e) e.blackUserId) _
            .WillCascadeOnDelete(False)
    End Sub
End Class

File: WebApplication/Models/Model1.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity;
using System.Linq;

namespace WebApplication
{
    public partial class Model1 : DbContext
    {
        public Model1()
            : base("name=Model1")
        {
        }

        public virtual DbSet<game> game { get; set; }
        public virtual DbSet<game_moves> game_moves { get; set; }
        public virtual DbSet<users> users { get; set; }

        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Entity<game>()
                .HasMany(e => e.game_moves)
                .WithRequired(e => e.game)
                .WillCascadeOnDelete(false);

            modelBuilder.Entity<users>()
                .HasMany(e => e.game)
                .WithRequired(e => e.users)
                .HasForeignKey(e => e.whiteUserId)
                .WillCascadeOnDelete(false);

            modelBuilder.Entity<users>()
                .HasMany(e => e.game1)
                .WithRequired(e => e.users1)
                .HasForeignKey(e => e.blackUserId)
                .WillCascadeOnDelete(false);
        }
    }
}
